crates/sui-sdk/src/types/framework.rs

```diff
@@ -39,7 +41,7 @@ impl<'a> Coin<'a> {
                     u64::from_le_bytes((&contents[ObjectId::LENGTH..]).try_into().unwrap());
 
                 Some(Self {
-                    coin_type,
+                    coin_type: Cow::Owned(coin_type.clone()),
```

Comment:
Hmm, not sure about this one. Cow::Borrowed seems not to work (complete noob with this Cow 😅 )